patches/node/enable_crashpad_linux_node_processes.patch

```diff
@@ -0,0 +1,51 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: VerteDinde <keeleymhammond@gmail.com>
+Date: Sun, 20 Nov 2022 21:45:20 -0800
+Subject: fix: enable crashpad for ELECTRON_RUN_AS_NODE linux processes
+
+Passes the crashpad handler PID and crashdump signal file descriptor
+to child processes spawned with `ELECTRON_RUN_AS_NODE` which is used
+by the crashpad client to connect with the handler process.
+
+diff --git a/lib/child_process.js b/lib/child_process.js
+index 73c1dc769542865bdf7a2a03c16cef141d3f4b05..76151c75ef7ad420d2642c1cd11c8624b7d7e2a0 100644
+--- a/lib/child_process.js
++++ b/lib/child_process.js
+@@ -59,6 +59,7 @@ let debug = require('internal/util/debuglog').debuglog(
+ );
+ const { Buffer } = require('buffer');
+ const { Pipe, constants: PipeConstants } = internalBinding('pipe_wrap');
++const { getCrashdumpSignalFD, getCrashpadHandlerPID } = process._linkedBinding('electron_common_crashpad_support');
+ 
+ const {
+   AbortError,
+@@ -159,7 +160,6 @@ function fork(modulePath, args = [], options) {
+       ArrayPrototypeSplice(execArgv, index - 1, 2);
+     }
+   }
+-
+   args = [...execArgv, modulePath, ...args];
+ 
+   if (typeof options.stdio === 'string') {
+@@ -613,6 +613,21 @@ function normalizeSpawnArguments(file, args, options) {
+                     'options.windowsVerbatimArguments');
+   }
+ 
++  if (process.platform === 'linux') {
++    if (ObjectPrototypeHasOwnProperty(options.env || process.env, 'ELECTRON_RUN_AS_NODE') &&
```

Comment:
Not blocking on it, but what's the path forward for this patch? Is it something we're anticipating floating for the long-term?

Reply:
I think there’s two approaches we could take in the long-term:

1. Move this functionality out of our forked child_process.fork and into the new UtilityProcess API. If we did that, we could likely remove this patch, but I think the API needs additional feature work before that can be done.
2. This becomes a long-running patch similar to our implementation of Breakpad in Node (see https://github.com/electron/electron/commit/b10f24386cb37d736b10f4831235eedcf1f0db4b )

I think the immediate plan is #2, but we’re looking at if #1 is feasible